pandas/core/dtypes/cast.py

```diff
@@ -232,7 +232,7 @@ def trans(x):
     return result
 
 
-def maybe_upcast_putmask(result, mask, other):
+def maybe_upcast_putmask(result: np.ndarray, mask: np.ndarray, other):
```

Comment:
can you type other here?

Reply:
looks like ill be doing a branch of just follow-ups soon